packages/clerk-js/src/ui/ImpersonationFab/ImpersonationFab.tsx

```diff
@@ -91,14 +95,70 @@ export const ImpersonationFab = () => {
   const session = useCoreSession();
   const { t } = useLocalizations();
   const { parsedInternalTheme } = useAppearance();
+  const containerRef = useRef<HTMLDivElement>(null);
   const actor = session?.actor;
   const isImpersonating = !!actor;
 
   //essentials for calcs
-  const top = '109px';
-  const right = '23px';
   const eyeWidth = parsedInternalTheme.sizes.$16;
   const eyeHeight = eyeWidth;
+  const topProperty = '--cl-impersonation-fab-top';
+  const rightProperty = '--cl-impersonation-fab-right';
+  const defaultTop = 109;
+  const defaultRight = 23;
+
+  const handleResize = () => {
+    const current = containerRef.current;
+    if (!current) {
+      return;
+    }
+
+    const outsideViewport =
+      current.offsetLeft < 0 ||
+      current.offsetLeft > window.innerWidth ||
+      current.offsetTop < 0 ||
+      current.offsetTop > window.innerHeight;
+
+    if (outsideViewport) {
+      document.documentElement.style.setProperty(rightProperty, `${defaultRight}px`);
+      document.documentElement.style.setProperty(topProperty, `${defaultTop}px`);
+    }
+  };
+
+  const onPointerDown: PointerEventHandler = () => {
+    window.addEventListener('pointermove', onPointerMove);
```

Comment:
⛔ Won't this function add the windowEventListener on every pointer down event?

Reply:
Yes. The pointermove is being removed on pointerup and the pointerup runs only once. Do you mean something else?

Reply:
Alternative: https://dev.to/js_bits_bill/addeventlistener-once-js-bits-565d